### ISSUE TITLE
Change JRA55-do data directory

### ISSUE THREAD

**Issue by aidanheerdegen** [COSIMA/access-om2#120]
The JRA55-do atmospheric dataset is currently used to force the ACCESS-OM2 models.

At the moment NCI users access this data in the  `ua8` directory where CMS (Paola) is downloading and updating the data.

JRA55-do is now an official dataset of input4MIPS, which is part of CMIP6. This means it is available under the `qv56` project and centrally managed by NCI.

This is a good thing. It will reduce Paola's workload, be downloaded automatically, and most importantly will be versioned, which was something that we discussed at TWG and definitely wanted.

Anyone who wishes to use this dataset will have to join `qv56`, which means anyone who wishes to run the ACCESS-OM2 models at NCI will also have to join.

We will need to determine the file paths. The data is here:

`/g/data/qv56/replicas/input4MIPs/CMIP6/OMIP/MRI/MRI-JRA55-do-1-3/`

So for example:

```
$ cd /g/data/qv56/replicas/input4MIPs/CMIP6/OMIP/MRI/MRI-JRA55-do-1-3/
$ ls atmos/3hr/rsds/gn/v20180412/rsds/
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_195801010130-195901010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_195901010130-196001010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_196001010130-196101010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_196101010130-196201010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_196201010130-196301010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_196301010130-196401010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_196401010130-196501010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_196501010130-196601010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_196601010130-196701010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_196701010130-196801010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_196801010130-196901010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_196901010130-197001010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_197001010130-197101010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_197101010130-197201010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_197201010130-197301010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_197301010130-197401010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_197401010130-197501010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_197501010130-197601010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_197601010130-197701010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_197701010130-197801010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_197801010130-197901010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_197901010130-198001010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_198001010130-198101010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_198101010130-198201010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_198201010130-198301010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_198301010130-198401010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_198401010130-198501010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_198501010130-198601010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_198601010130-198701010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_198701010130-198801010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_198801010130-198901010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_198901010130-199001010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_199001010130-199101010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_199101010130-199201010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_199201010130-199301010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_199301010130-199401010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_199401010130-199501010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_199501010130-199601010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_199601010130-199701010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_199701010130-199801010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_199801010130-199901010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_199901010130-200001010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_200001010130-200101010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_200101010130-200201010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_200201010130-200301010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_200301010130-200401010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_200401010130-200501010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_200501010130-200601010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_200601010130-200701010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_200701010130-200801010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_200801010130-200901010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_200901010130-201001010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_201001010130-201101010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_201101010130-201201010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_201201010130-201301010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_201301010130-201401010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_201401010130-201501010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_201501010130-201601010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_201601010130-201701010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_201701010130-201801010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_201801010130-201802020000.nc
```

This will need to be tested. I'll get around to it, but if someone else wants to put up their hands feel free and feedback to this issue so work is not duplicated.

**Comment by aekiss** [comment]
Should this be the JRA55-do path used for the default "vanilla" control directory repos? 
That would require all users to join qv56 - are there any restrictions on membership of this group?

Also @nichannah - will YATM require modification to handle these filenames? Or should we just make a lot of symlinks with names in the format we want?

**Comment by aidanheerdegen** [comment]
> Should this be the JRA55-do path used for the default "vanilla" control directory repos?

Yes.

> That would require all users to join qv56 - are there any restrictions on membership of this group?

Not that I know of.

> Also @nichannah - will YATM require modification to handle these filenames? Or should we just make a lot of symlinks with names in the format we want?

I was in the middle of testing this when other things cropped up. As a work-around for supporting the frankly annoying filenames I did as you suggested and made a bunch of links:

```
$ ls -l /g/data/ua8/JRA55-do/input4MIPs/ | head
total 3452
lrwxrwxrwx  1 aph502 ua8 176 Apr 12 14:41 areacellg.nc -> /g/data1b/qv56/replicas/input4MIPs/CMIP6/OMIP/MRI/MRI-JRA55-do-1-3/landIce/fx/areacellg/gn/v20180412/areacellg/areacellg_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn.nc
lrwxrwxrwx  1 aph502 ua8 174 Apr 12 14:41 areacello.nc -> /g/data1b/qv56/replicas/input4MIPs/CMIP6/OMIP/MRI/MRI-JRA55-do-1-3/ocean/fx/areacello/gn/v20180412/areacello/areacello_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn.nc
lrwxrwxrwx  1 aph502 ua8 184 Apr 12 14:41 friver_1958.nc -> /g/data1b/qv56/replicas/input4MIPs/CMIP6/OMIP/MRI/MRI-JRA55-do-1-3/ocean/day/friver/gn/v20180412/friver/friver_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_19580101-19590101.nc
lrwxrwxrwx  1 aph502 ua8 184 Apr 12 14:41 friver_1959.nc -> /g/data1b/qv56/replicas/input4MIPs/CMIP6/OMIP/MRI/MRI-JRA55-do-1-3/ocean/day/friver/gn/v20180412/friver/friver_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_19590101-19600101.nc
lrwxrwxrwx  1 aph502 ua8 184 Apr 12 14:41 friver_1960.nc -> /g/data1b/qv56/replicas/input4MIPs/CMIP6/OMIP/MRI/MRI-JRA55-do-1-3/ocean/day/friver/gn/v20180412/friver/friver_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_19600101-19610101.nc
lrwxrwxrwx  1 aph502 ua8 184 Apr 12 14:41 friver_1961.nc -> /g/data1b/qv56/replicas/input4MIPs/CMIP6/OMIP/MRI/MRI-JRA55-do-1-3/ocean/day/friver/gn/v20180412/friver/friver_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_19610101-19620101.nc
lrwxrwxrwx  1 aph502 ua8 184 Apr 12 14:41 friver_1962.nc -> /g/data1b/qv56/replicas/input4MIPs/CMIP6/OMIP/MRI/MRI-JRA55-do-1-3/ocean/day/friver/gn/v20180412/friver/friver_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_19620101-19630101.nc
lrwxrwxrwx  1 aph502 ua8 184 Apr 12 14:41 friver_1963.nc -> /g/data1b/qv56/replicas/input4MIPs/CMIP6/OMIP/MRI/MRI-JRA55-do-1-3/ocean/day/friver/gn/v20180412/friver/friver_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_19630101-19640101.nc
lrwxrwxrwx  1 aph502 ua8 184 Apr 12 14:41 friver_1964.nc -> /g/data1b/qv56/replicas/input4MIPs/CMIP6/OMIP/MRI/MRI-JRA55-do-1-3/ocean/day/friver/gn/v20180412/friver/friver_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_19640101-19650101.nc
```

It ain't pretty but it works. The current status is that I ran ok but did not reproduce results using the other JRA55 dataset. This is problematic.

**Comment by aekiss** [comment]
From Simon's email today it seems we should move to JRA55-do v1.4 but I can only find 1.3 in `/g/data/qv56/replicas` - can anyone suggest somewhere else to look? Or does it need to be downloaded?

**Comment by aidanheerdegen** [comment]
It will need to be downloaded, but there is a semi-automated process for that. I will consult with Paola.

**Comment by aekiss** [comment]
Thanks @aidanheerdegen

**Comment by aekiss** [comment]
I've lodged a help ticket at NCI to request download of 1.4.0.

**Comment by aekiss** [comment]
NCI has downloaded updates to JRA55-do here
/g/data/qv56/replicas/input4MIPs/CMIP6/OMIP/MRI/
including 1.3.2 and 1.4.0

**Comment by aekiss** [comment]
The naming convention has changed for 1.4.0:
```
ls /g/data/qv56/replicas/input4MIPs/CMIP6/OMIP/MRI/MRI-JRA55-do-1-4-0/ocean/day/tos/gn/v20190429/tos/
...
tos_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-4-0_gn_20120101-20121231.nc
tos_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-4-0_gn_20130101-20131231.nc
tos_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-4-0_gn_20140101-20141231.nc
tos_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-4-0_gn_20150101-20151231.nc
tos_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-4-0_gn_20160101-20161231.nc
tos_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-4-0_gn_20170101-20171231.nc
tos_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-4-0_gn_20180101-20181231.nc
tos_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-4-0_gn_20190101-20190105.nc
```

**Comment by aekiss** [comment]
Atmos data is more relevant:
```
ls /g/data/qv56/replicas/input4MIPs/CMIP6/OMIP/MRI/MRI-JRA55-do-1-4-0/atmos/3hr/rsds/gr/v20190429/rsds/
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-4-0_gr_195801010130-195812312230.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-4-0_gr_195901010130-195912312230.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-4-0_gr_196001010130-196012312230.nc
...
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-4-0_gr_201801010130-201812312230.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-4-0_gr_201901010130-201901052230.nc
```

**Comment by aekiss** [comment]
the time component of the filename is not the same for all files
```
cd /g/data/qv56/replicas/input4MIPs/CMIP6/OMIP/MRI/MRI-JRA55-do-1-4-0/atmos
for d in */*/*/*/*; do ls -1 $d | head -n 1; done
prra_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-4-0_gr_195801010130-195812312230.nc
prsn_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-4-0_gr_195801010130-195812312230.nc
rlds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-4-0_gr_195801010130-195812312230.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-4-0_gr_195801010130-195812312230.nc
huss_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-4-0_gr_195801010000-195812312100.nc
psl_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-4-0_gr_195801010000-195812312100.nc
tas_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-4-0_gr_195801010000-195812312100.nc
ts_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-4-0_gr_195801010000-195812312100.nc
uas_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-4-0_gr_195801010000-195812312100.nc
vas_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-4-0_gr_195801010000-195812312100.nc
areacella_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-4-0_gr.nc
sftof_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-4-0_gr.nc
```

**Comment by aekiss** [comment]
for comparison, here's 1.3.0:
```
cd /g/data/qv56/replicas/input4MIPs/CMIP6/OMIP/MRI/MRI-JRA55-do-1-3/atmos/
for d in */*/*/*/*; do ls -1 $d | head -n 1; done
prra_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_195801010130-195901010000.nc
prsn_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_195801010130-195901010000.nc
rlds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_195801010130-195901010000.nc
rsds_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_195801010130-195901010000.nc
huss_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_195801010000-195812312230.nc
psl_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_195801010000-195812312230.nc
tas_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_195801010000-195812312230.nc
ts_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_195801010000-195812312230.nc
uas_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_195801010000-195812312230.nc
vas_input4MIPs_atmosphericState_OMIP_MRI-JRA55-do-1-3_gn_195801010000-195812312230.nc
```

**Comment by aekiss** [comment]
I think it will cover all cases to have a new key `year+1`. How does this look? https://github.com/COSIMA/libaccessom2/pull/26

**Comment by aekiss** [comment]
the month and time part of the filename can be specified in [forcing.json](https://github.com/COSIMA/01deg_jra55_iaf/blob/master/atmosphere/forcing.json) which I expect (but haven't checked) will be the same for all files of a given field in a given version of JRA55-do, except perhaps for 2019 which is of course incomplete

**Comment by aekiss** [comment]
[make_remap_weights.py and make_remap_weights.sh](https://github.com/COSIMA/access-om2/tree/master/tools) specify `/g/data1/ua8/JRA55-do/` so I guess these should also be updated.

**Comment by aekiss** [comment]
scratch that thought - they're RYF files

**Comment by aekiss** [comment]
I've tested PR https://github.com/COSIMA/libaccessom2/pull/26 with these runs
1. https://github.com/COSIMA/1deg_jra55_iaf/tree/1deg_jra55_iaf_control - using `/g/data/ua8/JRA55-do/v1-3` and latest master https://github.com/COSIMA/libaccessom2/tree/2cc76e2
2. https://github.com/COSIMA/1deg_jra55_iaf/tree/1deg_jra55_iaf_test_yearp1 - using `/g/data/ua8/JRA55-do/v1-3` and the PR https://github.com/COSIMA/libaccessom2/pull/26 commit https://github.com/COSIMA/libaccessom2/tree/13315f5
3. https://github.com/COSIMA/1deg_jra55_iaf/tree/1deg_jra55_iaf_test_yearp1_qv56 - same as test 3 but using `/g/data/qv56/replicas/input4MIPs/CMIP6/OMIP/MRI/MRI-JRA55-do-1-3`
4. https://github.com/COSIMA/1deg_jra55_iaf/tree/1deg_jra55_iaf_test_yearp1_qv56_2 - same as test 3 but with forcing path specified via `config.yaml`
5. https://github.com/COSIMA/1deg_jra55_iaf/tree/1deg_jra55_iaf_test_yearp1_qv56_2_hybrid - same as test 4 but using `/g/data/ua8/JRA55-do/v1-3` for temperature and humidity

using these scripts
- https://github.com/aekiss/notebooks/blob/master/input4MIPs_testing.ipynb
- https://github.com/aekiss/notebooks/blob/master/check-MRI-JRA55-do-1-3.ipynb

Runs 1 and 2 are identical, as expected.
Run 3 and 4 are identical, but differ from 1 and 2.
Run 5 is identical to runs 1 and 2.

@aidanheerdegen also [noted above](https://github.com/COSIMA/access-om2/issues/120#issuecomment-518898904) that `/g/data/qv56/replicas/input4MIPs/CMIP6/OMIP/MRI/MRI-JRA55-do-1-3` gives different results from `/g/data/ua8/JRA55-do/v1-3`. 

This is because the near-surface temperature and humidity (`tas` and `huss`) are slightly different (as confirmed in run 5). Paola says `qv56` is a slightly newer version (`source_version = "1.3.1"` in .nc attributes) than `ua8` (`version = "v1.3"`), which explains it.

I've also confirmed that the month and time part of the filename is identical for all years (except the final one) in each variable we use in `/g/data/qv56/replicas/input4MIPs/CMIP6/OMIP/MRI/MRI-JRA55-do-1-3`, so the pattern used in [`forcing.json`](https://github.com/COSIMA/1deg_jra55_iaf/blob/1deg_jra55_iaf_test_yearp1_qv56_2/atmosphere/forcing.json) won't miss any v1.3 files (I haven't checked this for version 1.4 yet).

So in summary I'm confident that PR https://github.com/COSIMA/libaccessom2/pull/26 is ready to use.

**Comment by aekiss** [comment]
Remaining things to do in all IAF JRA55-do 1.3 configs:
- update all executables in `config.yaml` to use updated libaccessom2
- use this [`forcing.json`](https://github.com/COSIMA/1deg_jra55_iaf/blob/1deg_jra55_iaf_test_yearp1_qv56_2/atmosphere/forcing.json) and the atmosphere inputs in this [`config.yaml`](https://github.com/COSIMA/1deg_jra55_iaf/blob/1deg_jra55_iaf_test_yearp1_qv56_2/config.yaml)

**Comment by aekiss** [comment]
Closing. Both items above have now been done in the gadi-transition branch of all RYF and IAF configs  (not including the minimal 0.1 deg configs).
